subprojects/packagefiles/win-iconv/meson.build

```diff
@@ -0,0 +1,66 @@
+project(
+  'win-iconv',
+  'c',
+  version: '0.0.8',
+  meson_version: '>=1.1.0',
+)
+
+if host_machine.system() != 'windows'
+  error('win-iconv is only supported on windows, and not supported on '
+    + host_machine.system())
+endif
+
+cc = meson.get_compiler('c')
+
+c_args = []
+
+if cc.get_argument_syntax() == 'msvc'
+  c_args += '/D_CRT_SECURE_NO_WARNINGS'
+endif
+
+libiconv_dll_opt = get_option('libiconv_dll')
+
+if libiconv_dll_opt != ''
+  c_args += '-DUSE_LIBICONV_DLL'
```

Comment:
Upstream defines this unconditionally, which lets win-iconv respect the `WINICONV_LIBICONV_DLL` environment variable even if `DEFAULT_LIBICONV_DLL` isn't defined.  Let's define this by default, matching upstream.  If you want, you could add a Meson option to omit it.